src/main/webapp/script.js

```diff
@@ -0,0 +1,211 @@
+
+  const outputCanvas = createCanvasForId('output-canvas');
+  outputCanvas.setAttribute('width', CANVAS_WIDTH);
+  imagesContainer.append(outputCanvas);
+
+  // add sample image on page - original and blurred one.
+  putImageOnPage(SAMPLE_IMAGE_URL);
+});
+
+/**
+ * Function which is called when user chooses file to upload.
+ * @param {Event} event "Change" event on image upload button.
+ */
+async function handleImageUpload(event) {
+  await validateImageUpload().then(() => {
+    // if image is valid - put it on page
+    const imageUrl = URL.createObjectURL(event.target.files[0]);
+    putImageOnPage(imageUrl);
+  }).catch((error) => {
+    // if image is not valid - display error message
+    alert(error.message);
+  });
+}
+
+/**
+ * Function to validate the uploaded image.
+ * If valid - returns nothing, if not - throws error.
+ * @throws {Error} validation errors
+ */
+async function validateImageUpload() {
+  // get FileList with all the files from input element.
+  const files = document.getElementById('upload-image').files;
+
+  // check if no files were uploaded.
+  if (files.length === 0) {
+    throw new Error('Nothing is uploaded');
+  }
+
+  // get first (and the only) file from FileList.
+  const file = files[0];
+
+  // check whether file is jpeg or png.
+  const extension = await getExtensionIfSupported(file);
+
+  // check file size.
+  if (extension === 'png' && file.size > 8 * 1024 * 1024) {
+    throw new Error('Uploaded png file size can not exceed 8MB');
+  }
+
+  if (extension === 'jpeg' && file.size > 2 * 1024 * 1024) {
+    throw new Error('Uploaded jpeg file size can not exceed 2MB');
+  }
+
+  // File height or width can not be more than 1920px.
+  await validateImageDimensions(file);
+}
+
+/**
+ * Function to get file extension if it is png or jpeg
+ * and throw error otherwise
+ * @param {File} file
+ * @return {Promise}
+ */
+function getExtensionIfSupported(file) {
+  return new Promise(function(resolve, reject) {
+    const fileReader = new FileReader();
+
+    fileReader.onloadend = function(e) {
+      // get first 4 bytes from file - they contain information
+      // about the extension.
+      const bytes = new Uint8Array(e.target.result);
+
+      // convert those bytes to string
+      let header = '';
+      for (let i = 0; i < bytes.length; i++) {
+        header += bytes[i].toString(16);
+      }
+
+      const PNG_HEADERS = ['89504e47'];
+      if (PNG_HEADERS.includes(header)) {
+        resolve('png');
+      }
+
+      const JPEG_HEADERS = ['ffd8ffe0', 'ffd8ffe1',
+        'ffd8ffe2', 'ffd8ffe3', 'ffd8ffe8'];
+      if (JPEG_HEADERS.includes(header)) {
+        resolve('jpeg');
+      }
+
+      reject(new Error(
+          'Invalid file extension. Only jpeg and png images can be uploaded'));
+    };
+
+    // read first 4 bytes from file
+    fileReader.readAsArrayBuffer(file.slice(0, 4));
+  });
+}
+
+/**
+ * Function to make sure the file height and width are <= 1920px.
```

Comment:
I guess the image can also be vertically aligned, so it can be either 1920 x 1080, or 1080 x 1920. Or it can only be 1920 x 1080?
This is still incorrect, though.